cmd/utils/config.go

```diff
@@ -211,27 +213,70 @@ func SetP2PConfig(ctx *cli.Context, cfg *p2p.Config) {
 // method returns nil and an emphemeral key is to be generated.
 func setNodeKey(ctx *cli.Context, cfg *p2p.Config) {
 	var (
-		hex  = ctx.String(NodeKeyHexFlag.Name)
+		str  = ctx.String(NodeKeyHexFlag.Name)
 		file = ctx.String(NodeKeyFileFlag.Name)
 		key  *ecdsa.PrivateKey
 		err  error
 	)
 	switch {
-	case file != "" && hex != "":
+	case file != "" && str != "":
 		log.Fatalf("Options %q and %q are mutually exclusive", NodeKeyFileFlag.Name, NodeKeyHexFlag.Name)
 	case file != "":
 		if key, err = crypto.LoadECDSA(file); err != nil {
 			log.Fatalf("Option %q: %v", NodeKeyFileFlag.Name, err)
 		}
 		cfg.PrivateKey = key
-	case hex != "":
-		if key, err = crypto.HexToECDSA(hex); err != nil {
+	case str != "":
+		if key, err = crypto.HexToECDSA(str); err != nil {
 			log.Fatalf("Option %q: %v", NodeKeyHexFlag.Name, err)
 		}
 		cfg.PrivateKey = key
 	}
 }
 
+func LoadBlsNodeKey(ctx *cli.Context) (bls.SecretKey, error) {
+	if ctx.IsSet(NodeKeyFileFlag.Name) {
+		file := ctx.String(NodeKeyFileFlag.Name)
+		ecPriv, err := crypto.LoadECDSA(file)
+		if err != nil {
+			return nil, err
+		}
+		return bls.GenerateKey(crypto.FromECDSA(ecPriv))
+	}
+	if ctx.IsSet(NodeKeyHexFlag.Name) {
+		str := ctx.String(NodeKeyHexFlag.Name)
+		ecPriv, err := crypto.HexToECDSA(str)
+		if err != nil {
+			return nil, err
+		}
+		return bls.GenerateKey(crypto.FromECDSA(ecPriv))
+	}
+	if ctx.IsSet(BlsNodeKeyFileFlag.Name) {
+		file := ctx.String(BlsNodeKeyFileFlag.Name)
+		content, err := os.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+		if len(content) > 64 {
+			content = content[:64]
```

Comment:
The length of secret key must be 32 and its check is positioned inside of `SecretKeyFromByte()`. Could you clarify this?

Reply:
hex.DecodeString fails if I don't strip it. The same logic applies to parsing the ECDSA `nodekey` file. https://github.com/klaytn/klaytn/blob/v1.11.0/crypto/crypto.go#L176

But I will just use strings.TrimSpace for clarity.